### PR TITLE
Write chopper TDC time stamps serialised using the tdct flatbuffer schema 

### PR DESCRIPTION
**Pull request by SkyToGround**
### Issue

* Closes JIRA issue DM-1115

### Description of work

* Updated libraries to make the code compile on MacOSX.
* Added tdct writer module with tests.

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

**Comment by SkyToGround** [comment]
@matthew-d-jones, feel free to ask someone else in your team to review this pull request.

File: src/schemas/tdct/ChopperTimeStampWriter.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  auto Verifier = flatbuffers::Verifier(
+      reinterpret_cast<const std::uint8_t *>(Message.data()), Message.size());
+  return VerifytimestampBuffer(Verifier);
+}
+
+uint64_t
+ChopperTimeStampGuard::timestamp(FlatbufferMessage const &Message) const {
+  auto FbPointer = Gettimestamp(Message.data());
+  /// \todo This timestamp is currently EPICS epoch. This will have to be sorted
+  /// out.
```
Comment (mchorazak) [inline src/schemas/tdct/ChopperTimeStampWriter.cpp]: Create a jira ticket from \todo?

File: src/schemas/tdct/ChopperTimeStampWriter.cpp [hunk 2]
```diff
@@ -0,0 +1,124 @@
+/** Copyright (C) 2018 European Spallation Source ERIC */
+
+/// \file
+/// \brief Implement classes required for writing chopper time stamps.
+
+#include "../../helper.h"
+
+#include "ChopperTimeStampWriter.h"
+#include "HDFFile.h"
```
Comment (mchorazak) [inline src/schemas/tdct/ChopperTimeStampWriter.cpp]: Superfluous include statement

File: src/schemas/tdct/ChopperTimeStampWriter.cpp (hunk 1 above)
Comment (SkyToGround) [reply src/schemas/tdct/ChopperTimeStampWriter.cpp]: I believe we have several JIRA tickets on this issue (I created them). I have removed the comment.

File: src/schemas/tdct/ChopperTimeStampWriter.cpp (hunk 2 above)
Comment (SkyToGround) [reply src/schemas/tdct/ChopperTimeStampWriter.cpp]: Cleaned up the include statements.

File: src/tests/ChopperTimeStampWriterTests.cpp [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+  std::unique_ptr<tdct::ChopperTimeStampGuard> ReaderUnderTest;
+  static std::unique_ptr<std::int8_t[]> RawBuffer;
+  static size_t BufferSize;
+};
+std::unique_ptr<std::int8_t[]> ChopperTimeStampGuard::RawBuffer{nullptr};
+size_t ChopperTimeStampGuard::BufferSize{0};
+
+TEST_F(ChopperTimeStampGuard, GetSourceName) {
+  FileWriter::FlatbufferMessage TestMessage(
```
[the remark this answers is not in the document]
Comment (SkyToGround) [reply src/tests/ChopperTimeStampWriterTests.cpp]: Refactoring this reduced the file by two lines of code. :-)

**Comment by SkyToGround** [comment]
Thanks @mchorazak !

**Pull request merged**